packages/metadata/src/Decorated/storage/fromMetadata/createFunction.ts

```diff
@@ -219,7 +227,7 @@ export default function createFunction (registry: Registry, itemFn: CreateItemFn
     extendDoubleMap(registry, itemFn, storageFn);
   }
 
-  storageFn.keyPrefix = storageFn.iterKey || compactStripLength(storageFn())[1];
+  storageFn.keyPrefix = (arg?: any): Uint8Array => (storageFn.iterKey && storageFn.iterKey(arg)) || compactStripLength(storageFn())[1];
```

Comment:
```suggestion
  storageFn.keyPrefix = (arg?: any): Uint8Array =>
    (storageFn.iterKey && storageFn.iterKey(arg)) || compactStripLength(storageFn())[1];
```

Small readability change. (This and the previous one is hard to follow, just due to what we are dealing with here)